### PR TITLE
doc improvement: cleanup mesh changelog

### PR DESCRIPTION
**Pull request by mia-ko**
Minor edits for the mesh entries in the changelog

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -91,8 +91,8 @@ Bluetooth Mesh
   * :ref:`bt_mesh_ug_reserved_ids` with model ID and opcodes for the new :ref:`bt_mesh_le_pair_resp_readme` model.
   * :ref:`bt_mesh_light_ctrl_readme` APIs to match new Sensor APIs.
   * :ref:`ug_bt_mesh_configuring` with the recommendation on how to configure persistent storage to increase performance.
-  * :ref:`ug_bt_mesh_configuring` with the required configuration for the network core.
```
Comment (Balaklaka) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:94]: Why is this removed?

Comment (mia-ko) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:94]: aaah, not intentionally! Fixing it :)

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -435,64 +435,40 @@ Bluetooth Mesh samples
...
-  * Added:
-
-    * Support for pairing with display capability and the :ref:`bt_mesh_le_pair_resp_readme`.
-
-  * Fixed:
-
-    * An issue where the shell interface was not accessible over UART because UART was used as a transport for MCUmgr SMP protocol.
-      Shell is now accessible over RTT.
+  * Added support for pairing with display capability and the :ref:`bt_mesh_le_pair_resp_readme`.
+  * Fixed an issue where the shell interface was not accessible over UART because UART was used as a transport for MCUmgr SMP protocol.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:443]:
```suggestion
  * Fixed an issue where the shell interface was not accessible over UART because UART was used as a transport for the MCUmgr SMP protocol.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -78,21 +78,25 @@ Bluetooth Mesh
...
-  * :ref:`bt_mesh_ug_reserved_ids` with model ID and opcodes for the new :ref:`bt_mesh_le_pair_resp_readme` model.
-  * :ref:`bt_mesh_light_ctrl_readme` APIs to match new Sensor APIs.
-  * :ref:`ug_bt_mesh_configuring` with the recommendation on how to configure persistent storage to increase performance.
-  * :ref:`ug_bt_mesh_configuring` with the required configuration for the network core.
-  * Fixed an issue when the Time Server model rewrites TTL to zero forever during the unsolicited Time Status publication.
+    The old APIs based on the :c:struct:`sensor_value` type are deprecated, but are still available for backward compatibility and can be enabled for use by setting the :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` Kconfig option.
+  * The :ref:`bt_mesh_ug_reserved_ids` page with model ID and opcodes for the new :ref:`bt_mesh_le_pair_resp_readme` model.
+  * :ref:`bt_mesh_light_ctrl_readme` APIs to match new sensor APIs.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:92]:
```suggestion
  * :ref:`bt_mesh_light_ctrl_readme` APIs to match the new sensor APIs.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -435,64 +439,40 @@ Bluetooth Mesh samples
...
 
-  * Added:
-
-    * Support for the :ref:`zephyr:nrf52840dongle_nrf52840`.
+  * Added support for the :ref:`zephyr:nrf52840dongle_nrf52840`.
 
-  * Fixed:
-
-    * Building with :ref:`zephyr:sysbuild` for the :ref:`zephyr:nrf5340dk_nrf5340` and :ref:`zephyr:thingy53_nrf5340` boards.
+  * Fixed an issue when building with :ref:`zephyr:sysbuild` for the :ref:`zephyr:nrf5340dk_nrf5340` and :ref:`zephyr:thingy53_nrf5340` boards.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:459]: Should this issue be elaborated a bit more to know what the issue is?

**Review by SeppoTakalo — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 4 above)
Comment (PavelVPV) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:459]:
```
Fixed an issue where Bluetooth couldn't be initialized due to misconfiguration between the Bluetooth host and Bluetooth LE controller when building with :ref:`zephyr:sysbuild` for the :ref:`zephyr:nrf5340dk_nrf5340` and :ref:`zephyr:thingy53_nrf5340` boards
```

**Review by Balaklaka — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 4 above)
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:459]: This need to be changed for all of the places this is written.

**Review by divipillai — CHANGES_REQUESTED** [review]
One suggestion needs to be fixed.
